Expensify/App
# Start Using BeginSignIn command and remove obsolete API command calls

@robertjchen please review

### Details
Start using the `BeginSignIn` command added in https://github.com/Expensify/Web-Expensify/pull/34408

### Fixed Issues
$ https://github.com/Expensify/Expensify/issues/218743

### Tests
1. Do a normal sign-in flow for a user that already has a validated account, make sure it works properly:
https://user-images.githubusercontent.com/4741899/182488142-8044224b-396a-46b2-97a2-69599c0ee76d.mp4

---

1. With the same account do the same sign in. Click "Forgot?" on the password form.
2. Make sure you see the following:
![Screen Shot 2022-08-02 at 4 03 52 PM](https://user-images.githubusercontent.com/4741899/182488464-2b209c2a-5b44-43c4-bcd4-8172ae9874a8.png)

3. And that you get send a password reset link email.
4. Click the "Resend link" button.
5. Make sure you get another email

---

1. Sign in with an email that DOES NOT already have an account
2. Make sure that right away you see:
![Screen Shot 2022-08-02 at 4 14 01 PM](https://user-images.githubusercontent.com/4741899/182492001-a60bc05b-2a7a-4d2e-bc21-3fcda0eb188c.png)

3. ...And that you get an email with a sign-in link:
<img width="567" alt="Screen Shot 2022-08-02 at 4 14 25 PM" src="https://user-images.githubusercontent.com/4741899/182492061-f6ddfda2-edad-4ec7-8f27-8bef89129cf3.png">

4. Before validating that account by clicking the link in the email, sign-in with that account again, make sure that you get an email with a sign-in link again.

---

1. Sign-in with a phone number that does not have an account yet, make sure you see the following:
![Screen Shot 2022-08-02 at 4 16 39 PM](https://user-images.githubusercontent.com/4741899/182492268-777c98c2-6b47-4a16-9ead-6585d9b3336e.png)

2. And that you get a text message with your magic sign-in link:
![IMG_8397](https://user-images.githubusercontent.com/4741899/182492365-be1f253c-4170-48cb-9991-79eb7292acfb.PNG)

---
